assets/javascripts/discourse/components/chat-channel-row.js

```diff
@@ -17,6 +17,15 @@ export default Component.extend({
   active: propertyEqual("channel.id", "chat.activeChannel.id"),
   options: null,
 
+  init() {
+    this._super(...arguments);
+    this.options = this.options || {};
+    this.set(
+      "showSettingsButton",
+      this.options.settingsButton && this.currentUser.staff
```

Comment:
Is my understanding right that archive is the only option right now?

And later that options is controller by:
```
    this.siteSettings.chat_allow_archiving_channels &&
      !this.channel.isArchived &&
      !this.channel.isReadOnly
```

It means that when channel is currently archiving, then setting button is visible but nothing really happens on click. 

I see that later we will have option to mark channel as readonly and delete but maybe we should also check if any options are available to decide if button should be visible or not

Reply:
In the template for `chat-channel-settings-btn` I don't show the button if there are no options:

https://github.com/discourse-org/discourse-topic-chat/blob/b8266e0ad02c297b8e2c7c8138656f6179854542/assets/javascripts/discourse/templates/components/chat-channel-settings-btn.hbs#L1-L8